tests/infrastructure/test_docker_container.py

```diff
@@ -56,15 +60,25 @@ def mock_docker_client(monkeypatch):
     }
     fake_container.collection.get().attrs = attrs
     fake_container.attrs = attrs
-
+    fake_container.stop = MagicMock()
     # Return the fake container on lookups and creation
     mock.containers.get.return_value = fake_container
     mock.containers.create.return_value = fake_container
 
+    # Set attributes for infrastructure PID lookup
+    fake_api = MagicMock(name="APIClient")
+    fake_api.base_url = FAKE_BASE_URL
+    mock.api = fake_api
+
     monkeypatch.setattr("docker.from_env", MagicMock(return_value=mock))
     return mock
 
 
+@pytest.fixture
+def client_container_not_found(mock_docker_client):
+    mock_docker_client.containers.get.side_effect = [docker.errors.NotFound("msg")]
+
```

Comment:
I'm not sure it's worth moving this out into a fixture if it's a single line and only used once.